hack/lib/util.sh

```diff
@@ -435,12 +435,19 @@ kube::util::ensure_godep_version() {
 
   kube::log::status "Installing godep version ${GODEP_VERSION}"
   go install ./vendor/github.com/tools/godep/
-  GP="$(echo $GOPATH | cut -f1 -d:)"
-  hash -r # force bash to clear PATH cache
-  PATH="${GP}/bin:${PATH}"
+  if ! which godep >/dev/null 2>&1; then
```

Comment:
I don't like that there is inconsistency between the way we handle godep, and the way we are handling gazelle/kazel/misspell.

It's not enough of a dislike to block this PR that we badly need, but I'd really like if we could unify on a single way (maybe via a helper util function) to install a specific version of a go app from vendor, use it in the context of the script, with as little pollution of the environment as possible. Maybe I'll look at this in a follow up PR?

Reply:
We could set GOBIN in init.sh or something, and then always call $GOBIN/godep

But as long as we're checking the version of godep, it shouldn't matter.  I agree it would be cleaner to be consistent.  Our scripts are a mess here.

```
function gobin() {
  echo "${KUBE_OUTPUT_BINPATH}"
}

function go_install() {
  GOBIN="$(gobin)" go install "$@"
}

function run_private() {
  PATH="$(gobin):${PATH}" "$@"
}

function run_global() {
  "$@"
}


go_install ./vendor/github.com/tools/godep/
run_private godep save
```

alternatively, make teh callers of `ensure_godep_version` pass the path to the godep they are using, and they can pass `${GOBIN}/godep` ?

Agree it should be a followup